corehq/apps/reports/filters/api.py

```diff
@@ -197,3 +223,187 @@ def get(self, request, domain):
 
 class DeviceLogIds(DeviceLogFilter):
     field = 'device_id'
+
+
+@require_POST
+@toggles.COPY_CASES.required_decorator()
+@require_permission(HqPermissions.edit_data)
+@requires_privilege(privileges.CASE_COPY)
+@location_safe
+def copy_cases(request, domain, *args, **kwargs):
+    body = json.loads(request.body)
+
+    case_ids = body.get('case_ids')
+    if not case_ids:
+        return JsonResponse({'error': _("Missing case ids")}, status=400)
+
+    new_owner = body.get('owner_id')
+    if not new_owner:
+        return JsonResponse({'error': _("Missing new owner id")}, status=400)
+
+    censor_data = {
+        prop['name']: prop['label']
+        for prop in body.get('sensitive_properties', [])
+    }
+
+    case_copier = CaseCopier(
+        domain,
+        to_owner=new_owner,
+        censor_data=censor_data,
+    )
+    case_id_pairs, errors = case_copier.copy_cases(case_ids)
+    count = len(case_id_pairs)
+    return JsonResponse(
+        {'copied_cases': count, 'error': errors},
+        status=400 if count == 0 else 200,
+    )
+
+
+class CaseCopier:
+    """A helper class for copying cases."""
+    COMMCARE_CASE_COPY_PROPERTY_NAME = "commcare_case_copy"
```

Comment:
Nit: seeing this property without context I wasn't sure what it meant (it sounds like it might be a boolean).  Some other ideas after figuring it out:

* commcare_case_copy_of
* commcare_copy_of (we already know it's a case)
* commcare_original_case_id